### DISCUSSION TITLE
How do you create or boot ShredOS?

### DISCUSSION THREAD

**Discussion by PartialVolume** [PartialVolume/shredos.x86_64#205] [Polls]
Poll: In order to get an idea about what program or method is used to create a bootable ShredOS flash drive or boot ShredOS via other means.

**Comment by PartialVolume** [comment]
If you use some other method not listed, please let us know. Thanks.

**Reply by Firminator** [reply]
LinuxMint has a build-in utility called MintStick ( https://github.com/linuxmint/mintstick ) which is reliable, non-bloated, not Electron-based (like Etcher), and has a sha256 checksum and GPG verification. 
It handles both the ISO or USB IMG provided by ShredOS and I haven't had an issue ever with it or any other ISO or IMG.
For Windows I recommend Rufus when putting any one image on a flash drive. Pete is an expert in his field. The history of Rufus speaks for itself. Regular releases with bugfixes (fixing Windows bugs handling USB!) and expert knowledge since 11 years is hard to top.
Ventoy for Multi-images.

**Comment by tlschenkjr** [comment]
We use ShredOS in multiple ways including, dd to USB flash drive on LInux, creating a custom ISO and burning to CD, booting ShredOS from the PXE server that is part of RedHat Satellite.  The USB flash drive is used for wiping desktops, laptops, and one in a while, a rackmount server.  Boot from PXE is used for our server farm (10,000 plus rackmount servers and blade servers) and for virtual machines.   The reason why we customize ShredOS is because the download versions don't properly recognize drives in some of our HP servers (missing kernel menuconfig option) and to add support for log shipping via SFTP instead of lftp or tftp (both of which our security team frown upon) to the nwipe_launcher script..

**Reply by PartialVolume** [reply]
I would be interested in seeing some of those customisations. They might be useful in updating the ShredOS launcher especially with sftp.

**Reply by tlschenkjr** [reply]
I'll put them in a tree for you to look at.  They work, but it is not the most secure setup because I put the ssh keys in /etc/ssh_keys under the fsoverlay directory.  But I think that it would probably be better to get the key to the system running ShredOS using wget and a URL passed on the kernel command line.  I just haven't had the bandwidth to make those changes and test them.  As it stands, using sftp as the log shipping protocol requires creating a custom image with the SSH keys on the resulting ISO.  The other modification I made is to use dmidecode to put the system serial number into the name of the nwipe log file and the dmesg file, so that if you are running against multiple servers at the same time, they don't stomp on each other when writing to the remote host.

**Reply by PartialVolume** [reply]
That will be appreciated, thanks. Always good to see new ideas and it does look like it would be worth incorporating them into ShredOS.

**Reply by tlschenkjr** [reply]
It took me a while, but I finally uploaded my changes to ShredOS that we use here.  It is in a fork https://github.com/tlschenkjr/shredos.x86_64_SDSA.  The bulk of the changes are in nwipe_launcher, but there are a few more changes in the way ShredOS is configured and the archive shell script.

**Reply by PartialVolume** [reply]
Looks good, thanks. Are you ok if I cherry pick code from your fork, rather than try to merge the whole thing as there are changes to files not sftp related like additional programs, grub.cfg etc that I'd want to maybe separate out and review. There's a lot of tab changes so it makes it trickier see the actual code changes rather than formatting changes.

Hopefully I should be able get this into the next release, once I've had a good look at it and tested it.

**Reply by PartialVolume** [reply]
Also I'll most likely add those driver additions if you say they are required to recognise some drives on your HP servers.

**Reply by tlschenkjr** [reply]
You can use any part of it you like.  It's your project, I just wanted to show you what I had done for us to use it.  While I was adding the sftp code, it reworked some of the logic to use an if elif else structure instead of a if else with another if else embedded in it.

But do with it what you will.  The main thing I had to change about the kernel config was to enable CONFIG_PCI_MSI=y.  Without this, it would not recognize the HPE raid controllers or logical drives.

**Reply by PartialVolume** [reply]
Thanks. I've just committed that change "CONFIG_PCI_MSI=y". 52a07e3

**Reply by PartialVolume** [reply]
And also 96bb5cb. More to follow.

**Reply by Firminator** [reply]
More hardware support is very much appreciated.
Without having done any research on this particular driver I'm hopeful that the addition of "CONFIG_SCSI_MPI3MR=y" will enhance detection and support for a bunch of (newer?) RAID controllers similarly to what adding MPT2 and MEGARAID drivers has achieved in the past.

**Reply by PartialVolume** [reply]
@Firminator committed 3c9a12c

**Reply by PartialVolume** [reply]
These controllers are supported by the mpi3mr driver:
| Manufacturer | Model | Adapter type |
|----------------------|-----------|--------------------|
| Broadcom Ltd | SAS 4116 | Tri-Mode RAID Adapter |
| Broadcom Ltd | 9670W-16i 24G PCIe 4.0 | Tri-Mode RAID Adapter |
| Broadcom Ltd | 9670-24i 24G PCIe 4.0  | Tri-Mode RAID Adapter |
| Broadcom Ltd | 9660-16i 24G PCIe 4.0 | Tri-Mode RAID Adapter |
| Broadcom Ltd | 9620-16i 24G PCIe 4.0 | Tri-Mode RAID Adapter |
| Broadcom Ltd | 9600-24i 24G PCIe 4.0 | Tri-Mode RAID Adapter |
| Broadcom Ltd | 9600-16i 24G PCIe 4.0 | Tri-Mode RAID Adapter |
| Broadcom Ltd | 9600W-16e 24G PCIe 4.0 | Tri-Mode RAID Adapter |
| Broadcom Ltd | 9600-16e 24G PCIe 4.0 | Tri-Mode RAID Adapter |
| Broadcom Ltd | 9600-8i8e 24G PCIe 4.0 | Tri-Mode RAID Adapter |

**Reply by Firminator** [reply]
Tri-Mode. Wow.
Turns out Tri as in SATA, SAS, NVMe accd. to https://docs.broadcom.com/doc/BC00-0448EN
Looks like a winner for if you would custom-build a dedicated wiping box.
